girder/utility/acl_mixin.py

```diff
@@ -193,9 +201,162 @@ def prefixSearch(self, query, user=None, filters=None, limit=0, offset=0,
+            document, passed to MongoDB find() as the `projection` param.
+        :type fields: `str, list of strings or tuple of strings for fields to be included from the
+            document, or dict for an inclusion or exclusion projection`.
+        :param sort: The sort order.
+        :type sort: List of (key, order) tuples.
+        :param user: The user to check policies against.
+        :type user: dict or None
+        :param level: The access level.  Explicitly passing None skips doing
+            permissions checks.
+        :type level: AccessType
+        :param aggregateSort: A sort order to use if `sort` is None and an
+            aggregation is used.
+        :type aggregateSort: List of (key, order) tuples.
+        :returns: A pymongo Cursor, CommandCursor, or an iterable.  If a
+            CommandCursor, it has been augmented with a count function.
+        """
+        # If no user is specified and greater than READ access is requested,
+        # we won't return anything.  So as to always return a cursor for
+        # consistency, we make a query that will return no results
+        if user is None and level is not None and level > AccessType.READ:
+            query = {'__matchnothing': 'nothing'}
+        elif level is not None and (not user or not user['admin']):
+            # If the resourceColl isn't an access controlled model that we
+            # know how to reach, fall back to performing the ordinary query and
+            # then filtering it by permission.  For instance, if a model uses
+            # the acl mixin to get acl from a model that itself uses the acl
+            # mixin, this will return the correct results, but without the
+            # utility of being able perform count().
+            #  Note, this also handles models which use attachedToType and
+            # attachedToId, since self.model(None) will not be an access
+            # controlled model.
+            #  This is also the fall-back for Mongo < 3.4, as those versions do
+            # not support the aggregation steps that are used.
+            if (not isinstance(self.model(self.resourceColl), AccessControlledModel) or
+                    not getattr(self, '_dbserver_version', None) or
+                    getattr(self, '_dbserver_version', None) < (3, 4)):
+                return self._findWithPermissionsFallback(
+                    query, offset, limit, timeout, fields, sort, user, level,
+                    **kwargs)
+
+            query = query or {}
+            initialPipeline = [
+                {'$match': query},
+                {'$lookup': {
+                    'from': self.resourceColl,
+                    'localField': self.resourceParent,
+                    'foreignField': '_id',
+                    'as': '_access'
+                }},
+                {'$match': self.permissionClauses(user, level, '_access.')},
+            ]
+            countPipeline = initialPipeline[:] + [
```

Comment:
Why copy `initialPipeline` here and line 328?

Reply:
I'm going to assume I was either delusional or it was because I did something different a long time ago and failed to change it.